XGB 75 / XGB-92 architecture overview

fabianhauser opened the pull request.

raphiz commented on documents/final-submission-document/chapters/3_concept.tex:
```diff
@@ -10,6 +10,88 @@ \section{Architecture}
+We decided against the alternative SASL authentication method, SASL SCRAM~\cite{rfc7677}, that is also recommended by in the XMPP grid draft~\cite{ietf-mile-xmpp-grid-05}.
+As described in the corresponding architectural decision (Appendix~\ref{sec:architectural-decisions}), the main reason for SASL EXTERNAL is its higher level of security and and its relatively simple scaling capabilities.
+
+SASL EXTERNAL implies that the authentication takes place on a higher layer than the actual XMPP protocol. In our case, this implies authentication over TLS, i.e.~X.509 end-user certificates as specified in RFC6120~\cite{rfc6120}.
```
This section is a bit confusing...TLS is (in my perception) the layer below XMPP. Did I misunderstand this?

fabianhauser commented on documents/final-submission-document/chapters/2_analysis.tex:
```diff
@@ -80,5 +80,23 @@ \section{Requirements Analysis}
 
+\subsection{Domain Specific Language}
+
+Figure~\ref{fig:architecturedslgriddraft} and \ref{fig:architecturedslxeps} present an overview of most interactions between the different components as specified in the XMPP grid draft~\cite{ietf-mile-xmpp-grid-05} and as used in the referenced XEPs (see Chapter~\ref{sec:technical-background}).
```
- Formatting of IODEF could be a optional feature, but not in scope of this project
- Re JID: Add caption to graphics that they are simplified and show relevant interactions

fabianhauser commented:
@raphiz I addressed the issues you noted as discussed, please take a look if you find them satisfactory.
Also, I implemented [XGB-90](https://project.redbackup.org/browse/XGB-90), to note IODEF in the right place.
